### PR TITLE
Add more internal documentation for affine.jl

### PR DESCRIPTION
**Pull request by timholy**
@ChantalJuntao, does this help?

### CODE REVIEW

File: src/affine.jl [hunk 1]
```diff
@@ -1,79 +1,149 @@
...
+
+If `initial_tfm` is supplied, it will be composed with the one from `mat`.
+
+See also [`RegisterQD.aff`](@ref), which includes a translational component.
+"""
 function linmap(mat, img::AbstractArray{T,N}, initial_tfm=IdentityTransformation()) where {T,N}
     mat = [mat...]
     mat = reshape(mat, N,N)
     lm = LinearMap(SMatrix{N,N}(mat))
+    # The order below is subtle: you might think that `initial_tfm` should be supplied
```
Comment (ChantalJuntao) [inline src/affine.jl]: I didn't even remember to ask about this. This is super helpful.

File: src/affine.jl [hunk 2]
```diff
@@ -1,79 +1,149 @@
-#TODO more documentation?
+"""
+    tfm = linmap(mat, img, initial_tfm=IdentityTransformation())
+
+Construct a linear transformation from the values in `mat`.
+`mat` can be any list of `N^2` values, where `N` is the dimensionality.
+`img` is supplied solely to get the dimensionality, so that the size of the matrix in `tfm`
+is inferrable.
```
Comment (ChantalJuntao) [inline src/affine.jl]: The inferrabiliy is more about how the function works than how to use the function, so maybe it should go into the function as a comment instead?

File: src/affine.jl [hunk 3]
```diff
@@ -123,18 +193,15 @@ function qd_affine(fixed, moving, mxshift, linmins, linmaxs;
                                              SD = SD, minwidth=mw, initial_tfm=initial_tfm, thresh=thresh, print_interval=print_interval, kwargs...)
     print_interval < typemax(Int) && print("Running fine step\n")
     mw = mw./100
-    linmins, linmaxs = scalebounds(linmins, linmaxs, 0.5)
+    linmins, linmaxs = scalebounds(linmins, linmaxs, 0.5) # should these be narrowed further?
```
Comment (ChantalJuntao) [inline src/affine.jl]: What's the design logic behind using 0.5*linmins|linmax as the parameters for fine? Shouldn't the tfm from coarse get similar sized images within similar distances of the the final transformation, regardless of how far off the starting point was?

Comment (timholy) [reply src/affine.jl]: That's my sense too. We might change a number of things re affine, including shifting to more of a `Δc` parametrization (like in `default_minrot`). But let's do that in a separate PR.

**Pull request merged**